python/cudf_polars/cudf_polars/callback.py

```diff
@@ -174,16 +174,24 @@ def execute_with_cudf(
     device = config.device
     memory_resource = config.memory_resource
     raise_on_fail = config.config.get("raise_on_fail", False)
-    if unsupported := (config.config.keys() - {"raise_on_fail"}):
+    debug_mode = config.config.get("debug_mode", False)
+    if unsupported := (config.config.keys() - {"raise_on_fail", "debug_mode"}):
         raise ValueError(
             f"Engine configuration contains unsupported settings {unsupported}"
         )
     try:
         with nvtx.annotate(message="ConvertIR", domain="cudf_polars"):
+            translator = Translator(nt, debug_mode=debug_mode)
+            ir = translator.translate_ir()
+            if debug_mode and len(translator.errors):
+                unique_errors = set(translator.errors)
+                raise NotImplementedError(
+                    "Query contained unsupported operations", unique_errors
+                )
```

Comment:
*question*: We already have `raise_on_fail`. Do we separately need `debug_mode`? It seems like if `debug_mode` is `False` but we get some errors during translation then this condition swallows them and we would end up calling `nt.set_udf`?

Perhaps just:

```suggestion
            translator = Translator(nt)
            ir = translator.translate_ir()
            if len(translator.errors):
                unique_errors = sorted(set(translator.errors))
                raise NotImplementedError(
                    "Query contained unsupported operations", unique_errors
                )
```

And then `raise_on_fail` takes care of re-raising.

Or, since we now don't rely on exceptions for control flow:

```
translator = Translator(nt)
ir = translator.translate_ir()
if len(translator.errors):
    unique_errors = sorted(set(translator.errors))
    exception = NotImplementedError(..., unique_errors)
    if bool(int(os.environ.get("POLARS_VERBOSE", 0)):
        warnings.warn(...)
    if raise_on_fail:
        raise exception
else:
    nt.set_udf(
        partial(
            _callback,
            ir,
            device=device,
            memory_resource=memory_resource
        )
    )
```

WDYT?